internal/hub/config.go

```diff
@@ -93,7 +93,10 @@ func (c *Config) DecodeSession() (Session, error) {
 	}
 
 	hash := sha256.New()
```

Comment:
Just curious, why is a SHA-256 hash getting encrypted with AES here? Why not just AES encrypt the raw data directly?

Reply:
There shouldn't be any point in hashing *and* encrypting anything unless it happens to be a piece of a larger data set. For example you might encrypt a user profile database that hase various user info plus have the password hash in that data set, hence encrypting the hash. But the point isn't to encrypt the hash (which would be pointless by itself) but to keep the data together as a set.

I don't know what's going on here, but it ought to be apparent what the use cases is, this shouldn't be an open question while your swapping out hash mec

Reply:
It's a proof the client knows the secret without transmitting it in plain text, as it is also used for signing requests later. True, using a salt would be even more secure, but it's good enough for distributing maps keys. That's also why SHA-256 is used here and not SHA-1 as for file checksums.